File: CCC '21/J3.py
def solve(str,prev):
    a1,a2 = int(str[0]),int(str[1])
    a3 = int(str[2:])
    
    res = a1 + a2
    if res % 2==0 and res != 0:
        prev = "right"
    elif res % 2 == 1:
        prev = "left"
    print(prev,a3)
    return prev
    
N = input()
prev = ""

while N != "99999":
    prev = solve(N,prev)
    N = input()

File: CCC solutions/CCC '21/J3.cpp
#include <bits/stdc++.h>

void solve(std::string str, std::string& prev) {
    int a1 = str[0] - 48;
    int a2 = str[1] - 48;
    int a3 = std::stoi(str.substr(2,3));
    
    int res = a1 + a2;
    if (res % 2 == 0 && res != 0) { // even
        prev = "right"; 
    }else if (res % 2 == 1) {
        prev = "left";
    }
    std::cout << prev << " " << a3 << "\n";
} 

int main() {
    std::string N;
    std::string prev = "";
    std::cin >> N;
    while (N != "99999") {
        
        solve(N,prev);
        std::cin >> N;
    }
    return 0;
}